pkg/shares/merge_test.go

```diff
@@ -1,116 +0,0 @@
-package shares
-
-import (
-	"context"
-	_ "embed"
-	"encoding/json"
-	"testing"
-	"time"
-
-	"github.com/celestiaorg/celestia-app/pkg/appconsts"
-	"github.com/celestiaorg/celestia-app/test/util/testfactory"
-	"github.com/celestiaorg/rsmt2d"
-	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
-	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
-	coretypes "github.com/tendermint/tendermint/types"
-)
-
-func TestFuzz_merge(t *testing.T) {
-	t.Skip()
-	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
-	defer cancel()
-	for {
-		select {
-		case <-ctx.Done():
-			return
-		default:
-			Test_merge_randomData(t)
-		}
-	}
-}
-
-func Test_merge_randomData(t *testing.T) {
-	type test struct {
-		name      string
-		txCount   int
-		blobCount int
-		maxSize   int // max size of each tx or blob
-	}
-
-	tests := []test{
-		{"just one tx", 1, 0, 40},
-		{"just one blob", 0, 1, 40},
-		{"one of each random small size", 1, 1, 40},
-		{"one of each random large size", 1, 1, 400},
-		{"many of each random large size", 10, 10, 40},
-		{"many of each random large size", 10, 10, 400},
-		{"only transactions", 10, 0, 400},
-		{"only blobs", 0, 10, 400},
-	}
-
-	for _, tc := range tests {
-		t.Run(tc.name, func(t *testing.T) {
-			data := generateRandomBlockData(
-				tc.txCount,
-				tc.blobCount,
-				tc.maxSize,
-			)
-
-			shares, err := Split(data, false)
-			require.NoError(t, err)
-
-			eds, err := rsmt2d.ComputeExtendedDataSquare(ToBytes(shares), appconsts.DefaultCodec(), rsmt2d.NewDefaultTree)
-			assert.NoError(t, err)
-
-			got, err := merge(eds)
-			assert.NoError(t, err)
-			assert.Equal(t, data, got)
-		})
```

Comment:
[blocking]
iirc, I think we are keeping this purely because this is a good way to round trip test share splitting. (correct @rootulp?) perhaps we should add a comment to not delete this test (or merge)

if we decide to delete, then I think we should do so outside the scope of this PR. Does anyone have links to the old discussions deciding this?